extensions/amp-story-shopping/0.1/amp-story-shopping-attachment.js

```diff
@@ -113,49 +113,77 @@ export class AmpStoryShoppingAttachment extends AMP.BaseElement {
     // Update template on attachment state update or shopping data update.
     this.storeService_.subscribe(
       StateProperty.PAGE_ATTACHMENT_STATE,
-      (isOpen) => this.checkClearActiveProductData_(isOpen),
+      (isOpen) => this.onAttachmentStateUpdate_(isOpen),
       true /** callToInitialize */
     );
     this.storeService_.subscribe(
       StateProperty.SHOPPING_DATA,
-      (shoppingData) => {
-        if (this.isOnActivePage_()) {
-          this.checkOpenAttachment_(shoppingData);
-          this.updateTemplate_(shoppingData);
-        }
-      },
+      (shoppingData) => this.onShoppingDataUpdate_(shoppingData),
       true /** callToInitialize */
     );
   }
 
   /**
-   * Active product data is cleared after the attachment closes so that content does not jump.
+   * Handles clearing active product data and updating the template
+   * when there is no active product data.
    * @param {boolean} isOpen
    * @private
    */
-  checkClearActiveProductData_(isOpen) {
+  onAttachmentStateUpdate_(isOpen) {
+    if (!this.isOnActivePage_()) {
+      return;
+    }
+    // If template is closing, check to clear active product data and return early.
     if (!isOpen) {
-      Services.timerFor(this.win).delay(
-        () =>
-          this.storeService_.dispatch(Action.ADD_SHOPPING_DATA, {
-            'activeProductData': null,
-          }),
-        DRAGGABLE_DRAWER_TRANSITION_MS
```

Comment:
Having the same constant duplicated across two files is very fragile. If you don't want to listen for `animationEnd` events (which we should, for many reasons), please at least add a comment on both constants to remind developers to keep them in sync (i.e. if change this also change that)

Reply:
Great call. Was initially not wanting to do this because there are [two transitions on the element](https://github.com/ampproject/amphtml/blob/main/extensions/amp-story-page-attachment/0.1/amp-story-draggable-drawer.css#L13), resulting in two events being fired.

Using the `transitionend` definitely feels more easy to maintain and ensures the event is called when the drawer is fully closed. Happy to go that direction.